x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java

```diff
@@ -202,6 +203,8 @@ public String toString() {
     // Event
     public static class Event implements Writeable, ToXContentObject {
 
+        public static Event MISSING = new Event("", "", BytesArray.EMPTY, null);
```

Comment:
It would be better to distinguish missing events from actual events with a new flag in the serialization, but it would require a version increment, and it's probably not worth it.
An "empty" event (empty index, empty id, empty source) should be enough.

Reply:
If the `null` occurrence plus serialisation issue is rare, I guess having `MISSING` is ok. Otherwise, having a version gated behaviour might be acceptable.